operator/internal/central/extensions/reconcile_defaulting.go

```diff
@@ -0,0 +1,131 @@
+func FeatureDefaultingExtension(client ctrlClient.Client) extensions.ReconcileExtension {
+	return func(ctx context.Context, u *unstructured.Unstructured, _ func(extensions.UpdateStatusFunc), l logr.Logger) error {
+		return reconcileFeatureDefaults(ctx, client, u, l)
+	}
+}
+
+func reconcileFeatureDefaults(ctx context.Context, client ctrlClient.Client, u *unstructured.Unstructured, logger logr.Logger) error {
+	logger = logger.WithName("extension-feature-defaults")
+	if u.GroupVersionKind() != platform.CentralGVK {
+		logger.Error(errUnexpectedGVK, "unable to reconcile central", "expectedGVK", platform.CentralGVK, "actualGVK", u.GroupVersionKind())
+		return errUnexpectedGVK
+	}
+
+	if u.GetDeletionTimestamp() != nil {
+		logger.Info("skipping extension run due to deletionTimestamp being present on Central custom resource")
+		return nil
+	}
+
+	central := platform.Central{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &central); err != nil {
+		return errors.Wrap(err, "converting unstructured object to Central")
+	}
+
+	err := executeDefaultingFlows(ctx, logger, &central, client)
+	if err != nil {
+		return err
+	}
+
+	updatedObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&central)
+	if err != nil {
+		return errors.Wrap(err, "converting Central to unstructured object after extension execution")
+	}
+	u.Object = updatedObj
+
+	if err := platform.AddCentralDefaultsToUnstructured(u, &central); err != nil {
+		return errors.Wrap(err, "enriching unstructured Central object with defaults")
+	}
+
+	return nil
+}
+
+func executeDefaultingFlows(ctx context.Context, logger logr.Logger, central *platform.Central, client ctrlClient.Client) error {
+	origCentral := central.DeepCopy()
+
+	// This may update central.Defaults and central's embedded annotations.
+	for _, flow := range defaultingFlows {
+		if err := executeSingleDefaultingFlow(logger, central, client, flow); err != nil {
+			return err
+		}
+	}
+
+	// We persist the annotations immediately during (first-time) execution of this extension to make sure
+	// that this information is already persisted in the Kubernetes resource before we
+	// can realistically end up in a situation where reconcilliation might need to be retried.
+	newResourceVersion, err := patchCentralAnnotations(ctx, logger, client, origCentral, central.GetAnnotations())
+	if err != nil {
+		return errors.Wrap(err, "patching Central annotations")
+	}
+	central.SetResourceVersion(newResourceVersion)
+	return nil
+}
+
+func executeSingleDefaultingFlow(logger logr.Logger, central *platform.Central, client ctrlClient.Client, flow defaulting.CentralDefaultingFlow) error {
+	logger = logger.WithName(fmt.Sprintf("defaulting-flow-%s", flow.Name))
+	annotations := central.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	// to be thrown away after defaulting flow execution.
+	spec := central.Spec.DeepCopy()
+	status := central.Status.DeepCopy()
```

Comment:
Good idea. Perhaps it would make sense to mention explicitly in `CentralDefaultingFlow` declaration that mutations to `spec` and `status` are not preserved by design?